prepare-ocp4-aws-config.yml

```diff
@@ -1,56 +1,53 @@
-  hosts: all
+# Extra_vars for the running this playbook
+# 1. cluster_name: Name of the cluster to be provisioned
+# 2. aws_region: region on which the ocp_cluster needs to be deployed
+# 3. ocp_install_directory: Directory in which the cluster config files are generated
+#                           by default it is fetched from pipleline_repo_path
+# 4. ocp_install_config_template: ocp_config file template path
+# 5. vault_file_path: path to vault file. When given --vault-pass should be provided
+#                     over commandline 
+
+- name: "Prepare OCP 4.x configuration for installation on AWS"
+  hosts: localhost
   become: false
   gather_facts: false
-
-  vars_files:
-    - "{{ playbook_dir }}/../../../config/ansible/vault.yml"
-
-  vars:
-    cluster_name: "test-cluster"
-    aws_region: "us-east-2"
-
+  pre_tasks:
+    # set the default variable using set_fact
+    - name: "Set default varibles such that they can be overridden by extra-vars and environment variables"
+      set_fact:
+        cluster_name: "{{ cluster_name | default('test-cluster') }}"
+        aws_region: "{{ aws_region | default('us-east-2') }}"
+
+    - name: "Set OCP install directory"
+      # Note: all the set_fact variables are fetched from extra-vars passed to playbook
+      # if the variables are not found they would be assuming the default variables
+      set_fact:
+        ocp_install_directory: "{{ ocp_install_directory | default(playbook_dir) }}"
+        ocp_install_config_template: "{{ ocp_install_config_template |
+                                         default(playbook_dir+'/install-config.yaml.j2') }}"
+    - name: "include vault when given"
+      include_vars:
+        file: "{{ vault_file_path }}"
+      when: vault_file_path is defined
   tasks:
-    - set_fact:
-        ocp_install_directory: "{{ lookup('env', 'WORKSPACE') }}/operators/ocp-cluster-deployment/{{ cluster_name }}"
-
     - name: "Ensure that the ocp install directory exists and is empty"
       file:
-        path: "{{ ocp_install_directory }}"
+        path: "{{ ocp_install_directory }}/{{ cluster_name }}"
         state: "{{ item }}"
       with_items:
         - absent
         - directory
 
     - name: "Inject cluster name: {{ cluster_name }} and region: {{ aws_region }} into the install config"
       template:
-        src: "{{ lookup('env', 'WORKSPACE') }}/operators/ocp-cluster-deployment/install-config.yaml.j2"
-        dest: "{{ ocp_install_directory }}/install-config.yaml"
+        src: "{{ ocp_install_config_template }}"
+        dest: "{{ ocp_install_directory }}/{{ cluster_name }}/install-config.yaml"
 
-    - name: "Inject secrets into the install-config"
+    - name: "Inject secrets into the install-config when vault is defined"
       blockinfile:
-        path: "{{ ocp_install_directory }}/install-config.yaml"
+        path: "{{ ocp_install_directory }}/{{ cluster_name }}/install-config.yaml"
         block: "{{ cvp_vault.ocp_40_secrets | to_nice_yaml }}"
```

Comment:
While I like the idea of conditionally adding the secrets here, the users other than the CVP pipeline probably won't have the same structure for their pull secrets or ssh keys - it certainly would not be called cvp_vault. 
Expecting input files containing the secrets (e.g. pull_secret_path and ssh_key_path) and adding them to the end of the file separately (if they're supplied as extraVars) would be more generic and probably more usable for others.